src/hotspot/cpu/aarch64/register_aarch64.cpp

```diff
@@ -41,7 +41,7 @@ const char* RegisterImpl::name() const {
     "c_rarg0", "c_rarg1", "c_rarg2", "c_rarg3", "c_rarg4", "c_rarg5", "c_rarg6", "c_rarg7",
     "rscratch1", "rscratch2",
     "r10", "r11", "r12", "r13", "r14", "r15", "r16",
-    "r17", "r18", "r19",
+    "r17", NOT_R18_RESERVED("r18") WIN64_ONLY("rtls"), "r19",
```

Comment:
For me this line doesn't look good in case of expanding this functionality to macos-aarch64
as it's just the name of register and it's r18 on every platform except WIN64 and has nothing to do with reserving r18.

Reply:
The idea is that the naming should suggest that `r18` shouldn't be used on that particular platform. Same is true for macOS, but the ABI docs suggest a different usage, hence we have something like that in our internal branch for macOS:
```suggestion
    "r17", NOT_R18_RESERVED("r18") WIN64_ONLY("rtls") MACOS_ONLY("rplatform"), "r19",
```
Are you suggesting it should rather be something like this eventually?
```suggestion
    "r17", LINUX_ONLY("r18") WIN64_ONLY("rtls") MACOS_ONLY("rplatform"), "r19",
```

Reply:
this looks better I think, if it's done right from beginning, we won't have to modify it later.
The Question is, can we do it ahead of JEP-391 ?
If we can't then maybe better to leave it this way for now:
WIN64_ONLY("rtls") NOT_WIN64("r18")

Reply:
Let us go with the following in this PR as that's the extend of this PR:
```suggestion
    "r17", LINUX_ONLY("r18") WIN64_ONLY("rtls"), "r19",
```
We can update it accordingly in a PR for JEP-391.

Reply:
Ok, Great.